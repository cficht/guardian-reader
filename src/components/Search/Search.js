import React from 'react';
import PropTypes from 'prop-types';
import './Search.scss';

export default function Search({ searchText, handleSearchText, handleSearch, handlePerPage }) {
  return (
    <section className="search-section">
      <form onSubmit={handleSearch}>
        <div className="search-div">
          <input type="text" value={searchText} onChange={handleSearchText}/>
          <button>Search</button>
        </div>
        <div className="per-page-div">
          <label>Articles Per Page:
            <select onChange={handlePerPage}>
              <option>10</option>
              <option>20</option>
              <option>50</option>
              <option>100</option>
            </select>
          </label>   
        </div> 
      </form>
    </section>
  );
}

Search.propTypes = {
  searchText: PropTypes.string,
  handleSearchText: PropTypes.func,
  handleSearch: PropTypes.func,
  handlePerPage: PropTypes.func
};
