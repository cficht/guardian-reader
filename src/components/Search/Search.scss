$primary-color: #052962;
$secondary-color: #FFFFFF;
$third-color: whitesmoke;
$fourth-color: #00417B;
$fifth-color: #B2C9D9;

.search-section {
  display: flex;
  justify-content: space-evenly;
  min-width: 50%;
  margin: 20px;
  background-color: $third-color;

  form {
    width: 100%;
  }
}

.search-div {
  display: flex;
  justify-content: center;
  width: 100%;
  padding: 5px 0px 0px 0px;

  input {
    margin: 5px 10px 5px 10px;
  }

  button {
    margin: 5px 10px 5px 10px;
    padding: 5px 20px 5px 20px;
    text-align: center;
    color: $secondary-color;
    background-color: $primary-color;
  }

  button:hover {
    background-color: $fourth-color;
  }
}

.per-page-div {
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  padding: 0px 0px 5px 0px;

  select {
    margin: 5px 10px 5px 10px;
  }
}