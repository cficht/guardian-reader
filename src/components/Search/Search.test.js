/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Search from './Search';

test('renders Search.js correctly', () => {
  const tree = renderer
    .create(<Search />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
