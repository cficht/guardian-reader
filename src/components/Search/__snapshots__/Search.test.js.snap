// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Search.js correctly 1`] = `
<section
  className="search-section"
>
  <form>
    <div
      className="search-div"
    >
      <input
        type="text"
      />
      <button>
        Search
      </button>
    </div>
    <div
      className="per-page-div"
    >
      <label>
        Articles Per Page:
        <select>
          <option>
            10
          </option>
          <option>
            20
          </option>
          <option>
            50
          </option>
          <option>
            100
          </option>
        </select>
      </label>
    </div>
  </form>
</section>
`;
